libbeat/outputs/redis/redis_test.go

```diff
@@ -0,0 +1,96 @@
+package redis
+
+import (
+	"testing"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/outputs"
+	_ "github.com/elastic/beats/libbeat/outputs/codec/json"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMakeRedis(t *testing.T) {
+	tests := []struct {
+		Name                 string
+		Config               map[string]interface{}
+		Valid                bool
+		AdditionalValidation func(*testing.T, outputs.Group)
+	}{
```

Comment:
The table is okay as is. Just some tips:
- prefer `map[string]struct{ ... }` over `[]struct { Name string }` (still used by older code). This makes the table a little more dense and randomizes execution order.
- if you get tired of writing `map[string]interface{}` over and over again (I do get tired :) ) create a type like `type config map[string]interface{}`.
- when writing the table use struct keys and don't mention optional settings. This helps if table become large. e.g.
- if checks are based on functions but become verbose+repetitive consider to provide combinators.

e.g.

```
tests := map[string]struct{
  config config
  invalid bool
  checks checker
}{
  "No host": {
    config: config{
      "hosts": []string{},
    },
    invalid: true,
  },
  ...
  "Specific and default password": {
    config: config{
      "hosts":    []string{"redis://localhost:6379", "rediss://:mypassword@localhost:6380"},
      "password": "defaultPassword",
    },
    checks: checks(numclients(2), clientPassword(0, "defaultPassword"), clientPassword(1, "mypassword"))
  },
}
```

with:

```
type checker func(*testing.T, outputs.Group)
checks := func(cs ...checker) checker {
  return func(t *testing.T, g outputs.Group) {
    for _, c := range cs {
      c(t, g)
    }
  }
}

numCients := func(n int) checker {
  return func(t *testing.T, g outputs.Group) {
    require.Equal(t, n, g.Clients)
  }
}
```

Reply:
Really helpfull tips thanks